docs/19_security_audit_document.md

```diff
@@ -0,0 +1,275 @@
+
+# Goals
+> Identify success metrics and measurable goals.
+
+This document aims the following goals: 
+
+- Explain the security aspects of the Metrics Web App application.
+- Explain the security aspects of the CI Integrations.
+- Explain the security aspects of the Coverage Converter.
+
+# Non-Goals
+> Identify what's not in scope.
+
+This document does not describe the implementation of the Metrics project parts.
+
+# Table of Contents
+
+1. [Metrics Web Application security aspects](#metrics-web-application)
+2. [CI Integrations security aspects](#ci-integrations)
+3. [Coverage converter security aspects](#coverage-converter)
+
+# Metrics Web Application
+> Describe the security aspects of the Metrics Web Application.
+
+Metrics Web Application uses Firebase services. Consider the following documentation that describes the [`Privacy and Security of Firebase services`](https://firebase.google.com/support/privacy).
+
+## Authentication
+
+Metrics Web Application uses the `Email and Password` and the `Google` sign-in options.
+
+The `Email and Password` sign in option uses the [`firebase_auth`](https://pub.dev/packages/firebase_auth) package that is a part of [`FlutterFire`](https://firebase.flutter.dev/). `FlutterFire` is a set of official Firebase plugins for Flutter.
+
+The `Google` sign in option uses the [google_sign_in](https://pub.dev/packages/google_sign_in) package made by the Flutter Team.
+
+The `Google` sign in option provides the user's email, OAuth2 access token, and OpenID Connect ID token, which then are passed to the [Firebase `Social Authentication` (3-rd party authentication method)](https://firebase.flutter.dev/docs/auth/social).
+
+Firebase saves the information about users including the email, sign-in provider, creation date, last sign in and the user's UID, and is available in the [`Firebase Console`](https://console.firebase.google.com/).
+
+`Firebase Authentication` persists the user's authentication state by default using the `localstorage`. The same strategy is used in Metrics Web Application. Consider the following links to learn more about the authentication state persistence: [`FlutterFire` documentation](https://firebase.flutter.dev/docs/auth/usage#persisting-authentication-state) and [`Firebase` documentation](https://firebase.google.com/docs/auth/web/auth-state-persistence).
+
+## Database
+
+Metrics Web Application uses the `Firebase Cloud Firestore` as a database.
+
+Consider the following [document that describes the  data model used in `Firebase Cloud Firestore`](https://firebase.google.com/docs/firestore/data-model).
+
+### The `allowed_email_domains` collection
+
+The `allowed_email_domains` collection contains the allowed email domains for the `Google` sign in option.
+
+#### Documents
+
+Each document represents a single valid email domain.
+
+#### Document structure
+
+Each document stores a valid email domain as a name and does not contain any fields.
```

Comment:
```suggestion
### The `allowed_email_domains` collection

The `allowed_email_domains` collection contains the allowed email domains for the `Google Sign-In` method. The single document of this collection stands for the email domain that is allowed and may be used to authorize a user.

The documents of this collection do not have any fields. Instead, the ID of the document represents a single email domain. This prevents duplicates of email domains and simplifies validating as documents aren't to be fetched. 

Here is a table of security rules applied to the `allowed_email_domains` collection.
... ... 
```
Maybe it would be better to avoid creating a vast of subsections with little content. Also, Placing security rules after collection description seems to make sense. Please apply to all the collections-related sections.